phi/torch/nets.py

```diff
@@ -380,27 +381,180 @@ def forward(self, x):
 
         return out
 
+class Dense_ResNet_Block(nn.Module):
+
+    def __init__(self, in_channels, mid_channels, batch_norm, activation):
+        super(Dense_ResNet_Block, self).__init__()
+
+        self.bn1 = NORM[1](mid_channels) if batch_norm else nn.Identity()
+        self.linear1 = nn.Linear(in_channels, mid_channels)
+
+        self.bn2 = NORM[1](in_channels) if batch_norm else nn.Identity()
+        self.linear2 = nn.Linear(mid_channels, in_channels)
+
+    def forward(self, x):
+        x = TORCH.as_tensor(x)
+        out = self.activation()(self.bn1(self.linear1(x)))
+
+        out = self.activation()(self.bn2(self.linear2(out)))
+
+        out = out + x
+
+        return out
+
+def get_mask(inputs, reverse_mask, data_format = 'NHWC'):
+    shape = inputs.shape
+    if len(shape) == 2:
+        N = shape[-1]
+        range_n = torch.arange(0, N)
+        even_ind = range_n % 2
+        checker = torch.reshape(even_ind, (-1, N))
+    elif len(shape) == 4:
+        H = shape[2] if data_format == 'NCHW' else shape[1]
+        W = shape[3] if data_format == 'NCHW' else shape[2]
+
+        range_h = torch.arange(0, H)
+        range_w = torch.arange(0, W)
+
+        even_ind_h = range_h % 2
+        even_ind_w = range_w % 2
+
+        ind_h = even_ind_h.unsqueeze(-1).repeat(1, W)
+        ind_w = even_ind_w.unsqueeze( 0).repeat(H, 1)
+
+        checker = torch.logical_xor(ind_h, ind_w)
+
+        checker = checker.reshape(1, 1, H, W) if data_format == 'NCHW' else checker.reshape(1, H, W, 1)
+        checker = checker.long()
+
+    else:
+        raise ValueError('Invalid tensor shape. Dimension of the tensor shape must be '
+                         '2 (NxD) or 4 (NxCxHxW or NxHxWxC), got {}.'.format(inputs.get_shape().as_list()))
+
+    if reverse_mask:
+        checker = 1 - checker
+
+    return checker.to(TORCH.get_default_device().ref)
+
+class Coupling_layer(nn.Module):
```

Comment:
Please stick to the Python naming conventions, this should be `CouplingLayer`

Reply:
Ok! I've corrected the class definitions.